hq/markdown/snyk.md

```diff
@@ -0,0 +1,227 @@
+# Snyk
+
+## Introduction
+
+Snyk is a managed service which uses a dependency analysis tool and a vulnerability database to inform you of
+vulnerabilities in your application.
+
+## Installing
+
+Snyk requires node, and `npm`.  To install on a mac, please use:
+
+```
+brew install npm
+npm install -g snyk
+```
+
+Snyk then requires an authentication token to be set up.  Github auth is almost certainly most appropriate.
+
+```
+snyk auth
+```
+
+# Usage
+
+## Typical Usage for Applications
+
+In the build process (eg TeamCity), you can add a build step to check with the Snyk database for vulnerabilities.
+
+This information can be simply treated as reporting information, or can be used to fail a build if the return
+code is non-zero (ie at least one vulnerability found or error).
+
+## Use with Scala
+
+### Pre-requisites
+
+Snyk needs to first create a dependency tree for your project (based on the sbt dependencies), then check with the
+Snyk database.
+
+Dependency graph creation is via an sbt plugin.  Add the following line to your project plugins file or your `.sbt/plugins`
+directory:
+
+
+```
+/Users/<you>/.sbt/<sbt-version>/plugins/snyk.sbt
+or
+<projectroot>/project/plugins.sbt
+```
+
+```
+addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.9.0")
+```
+
+### Invocation
+
+It is recommended that `sbt` itself is invoked first, to confirm that the build file is good.  You
+can then invoke `snyk` as follows.  If you are working on a new snyk implementation, then it is often
+worth cutting down the on-screen noise by using `--show-vulnerable-paths=false`.
+
+```
+sbt test
+snyk test --file=build.sbt [--show-vulnerable-paths=(true|false)]
+```
+
+### Gotchas
+
+#### With SBT 1.0.0+
+
+It appears that sbt expects to invoke `dependency-tree`.  After sbt 1.0, this command appears to be `dependencyTree`.
```

Comment:
Be good to raise this issue with Snyk if we haven't already!

Reply:
I've briefly discussed this with Kate and we will do so.

Reply:
Added to Trello backlog